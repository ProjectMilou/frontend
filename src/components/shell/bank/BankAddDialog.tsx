import React from 'react';
import { Box, Button, Typography } from '@material-ui/core';
import { useTranslation } from 'react-i18next';
// TODO: Screenshot the webform checkbox
import ImgInstruction from '../../../assets/images/analyse.png';

interface BankAddDialogProps {
  link: string;
}

const BankAddDialog: React.FC<BankAddDialogProps> = (props) => {
  const { link } = props;
  const { t } = useTranslation();

  return (
    <Box my={3}>
      <Typography align="center" variant="h5" gutterBottom>
        {t('shell.bank.add.dialog.headline')}
      </Typography>
      <Typography align="center" variant="body1" gutterBottom>
        {t('shell.bank.add.dialog.subheadline')}
      </Typography>
      <img
        width="100%"
        src={ImgInstruction}
        alt="Make sure to click the checkbox!"
      />

      <Button variant="contained" color="primary" fullWidth href={link}>
        {t('shell.bank.add.dialog.toWebform')}
      </Button>
    </Box>
  );
};

export default BankAddDialog;
