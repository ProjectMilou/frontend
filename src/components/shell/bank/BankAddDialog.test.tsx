import * as React from 'react';
import { fireEvent, render, waitFor } from '@testing-library/react';
import BankAddDialog from './BankAddDialog';

describe('BankAddDialog', () => {

  const linkProp = "https://www.example.com"

  test('should match snapshot', () => {
    const { container } = render(<BankAddDialog link={linkProp}/>);
    expect( container ).toMatchSnapshot();
  });

  test('should open link', async () => {
    const {getByRole} = render(<BankAddDialog link={linkProp}/>);
    const link = getByRole("link")
    expect(link).toBeInTheDocument()
    global.open = jest.fn();
    fireEvent.click(link);
    waitFor(() => {
      expect(global.open).toBeCalled();
    }).then(() => {})
  });
});