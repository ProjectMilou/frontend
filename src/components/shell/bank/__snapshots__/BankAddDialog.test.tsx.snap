// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BankAddDialog should match snapshot 1`] = `
<div>
  <div
    class="MuiBox-root MuiBox-root-2"
  >
    <h5
      class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom MuiTypography-alignCenter"
    >
      shell.bank.add.dialog.headline
    </h5>
    <p
      class="MuiTypography-root MuiTypography-body1 MuiTypography-alignCenter"
    >
      shell.bank.add.dialog.subheadline
    </p>
    <img
      alt="shell.bank.add.dialog.subheadline"
      class="makeStyles-instruction-1"
      src="instructions.png"
    />
    <a
      aria-disabled="false"
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-fullWidth"
      href="https://www.example.com"
      tabindex="0"
      target="_blank"
    >
      <span
        class="MuiButton-label"
      >
        shell.bank.add.dialog.toWebform
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </a>
  </div>
</div>
`;
